src/python/pants/engine/target.py

```diff
@@ -2692,6 +2692,26 @@ def __init__(
         object.__setattr__(self, "exclude", FrozenOrderedSet(sorted(exclude)))
 
 
+@union(in_scope_types=[EnvironmentName])
+@dataclass(frozen=True)
+class TransitivelyExcludeDependenciesRequest(Generic[FS], EngineAwareParameter):
+    """A request to transitvely exclude dependencies of a "root" node.
+
+    This is similar to `InferDependenciesRequest`, except the request is only made for "root" nodes
+    in the dependency graph.
+
+    This mirrors the public facing "transitive exclude" dependency feature (i.e. `!!<address>`).
+    """
+
+    infer_from: ClassVar[Type[FS]]  # type: ignore[misc]
```

Comment:
Should this be `infer_from`? I don't have a better name and it matches the inference request

Reply:
I don't have a better suggestion either.

However, I'd perhaps suggest moving this part:
```python
request_type(
    request_type.infer_from.create(tgt)
)
```
into a helper method on this class, to make it more clear and keep the use of the field more local to where it is declared, and makes the code easier to read/understand in `graph.py`, I find.

Reply:
This part is a copy+paste-rename from `resolve_dependencies`'s:

```python
            Get(
                InferredDependencies,
                {
                    inference_request_type(
                        inference_request_type.infer_from.create(tgt)
                    ): InferDependenciesRequest,
                    environment_name: EnvironmentName,
                },
            )
```

Reply:
doesn't mean it couldn't be improved.. ;)